lib/block-supports/layout.php

```diff
@@ -233,3 +233,36 @@ function( $matches ) {
 	remove_filter( 'render_block', 'wp_restore_group_inner_container', 10, 2 );
 }
 add_filter( 'render_block', 'gutenberg_restore_group_inner_container', 10, 2 );
+
+
+/**
+ * For themes without theme.json file, make sure
+ * to restore the outer div for the aligned image block
+ * to avoid breaking styles relying on that div.
+ *
+ * @param string $block_content Rendered block content.
+ * @param array  $block         Block object.
+ * @return string Filtered block content.
+ */
+function gutenberg_restore_image_outer_container( $block_content, $block ) {
+	$image_with_align = '/(^\s*<figure\b[^>]*)wp-block-image\s([^"]*((alignleft)|(alignright)|(aligncenter))(\s|")[^>]*>((.|\S|\s)*)<\/figure>)/U';
+
+	if (
+		'core/image' !== $block['blockName'] ||
+		WP_Theme_JSON_Resolver::theme_has_support() ||
+		0 === preg_match( $image_with_align, $block_content )
+	) {
+		return $block_content;
+	}
+
+	$updated_content = preg_replace_callback(
+		$image_with_align,
+		static function( $matches ) {
+			return '<div class="wp-block-image">' . $matches[1] . $matches[2] . '</div>';
+		},
+		$block_content
+	);
+	return $updated_content;
+}
+
+add_filter( 'render_block', 'gutenberg_restore_image_outer_container', 10, 2 );
```

Comment:
Was there a reason for not using `render_block_core/image` as the filter?

Reply:
No particular reason, I guess my reasoning is that this change is tied to the "layout support" so I put here similarly to the previous one we have for group blocks. (same use-case)

Reply:
Or maybe you mean why not use the `render_block_core/image` filter name? If that's the case, I just didn't know that these block specific filters exist.

Reply:
Yep, sorry for not being clear here. The filter was introduced in 5.7, see https://core.trac.wordpress.org/changeset/50123. It basically allows you to skip the `'core/image' !== $block['blockName']` check and only runs if the block is actually used on a page.

Reply:
Nice 👍 We should probably use it for both image and group filters.

Reply:
Opened a follow-up here https://github.com/WordPress/gutenberg/pull/38657